### PR TITLE
add gpu naive conv for verification backend

### PR DESCRIPTION
**Pull request by carlushuang**

### CODE REVIEW

File: script/smoke_test.sh [hunk 1]
```diff
@@ -32,7 +33,7 @@ EXE=./out/conv_driver.exe
 batch_size=( 2 )
 image_size=( 14 32 55 )
 channel_size=( 64 32 )
-group_size=( 1 2 4 )
+group_size=( 1 )
```
Comment (shaojiewang) [inline script/smoke_test.sh:36]: why remove multi-group check？

File: driver/gpu_naive_conv.h [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+
+static inline void gpu_naive_conv_wrw_nchw_fp16(void *src, void *filter,
+                                       void *dst, size_t n, size_t w, size_t h,
+                                       size_t c, size_t k, size_t fx, size_t fy,
+                                       size_t px, size_t py, size_t sx,
+                                       size_t sy, size_t dx, size_t dy, size_t group)
+{
+    size_t ho = gpu_naive_conv_out_size(h, py, dy, fy, sy);
+    size_t wo = gpu_naive_conv_out_size(w, px, dx, fx, sx);
+    size_t k_per_group  = k / group;
```
Comment (shaojiewang) [inline driver/gpu_naive_conv.h:370]: Did we check if group is greater than zero?

Comment (carlushuang) [reply driver/gpu_naive_conv.h:370]: No here... let me add

File: script/smoke_test.sh (hunk 1 above)
Comment (carlushuang) [reply script/smoke_test.sh:36]: remove by accedent. let me add

**Comment by fronteer** [comment]
Suggest to add hipDeviceSynchronize() or hipStreamSynchronize(0) between gpu_naive_conv_fwd/bwd/wrw_nchw_fp32() and the following hipMemcpy(),  because without synchronizing, there is no guarantee that the kernel is finished while the copying is done since hipMemcpy() is a synchronous copying, which internally use different queue as the stream dispatching the kernel.  

You can also use  hipMemcpyAsync() to enforce an ordering by let the copying to use the same stream as the kernel

**Review by shaojiewang — DISMISSED** [review]

**Review by shaojiewang — APPROVED** [review]

**Pull request merged**